docs/reference/filebeat/filebeat-input-aws-s3.md

```diff
@@ -421,6 +421,11 @@ This sets the maximum number of JavaScript VM sessions that will be cached to av
 The maximum duration that an SQS `ReceiveMessage` call should wait for a message to arrive in the queue before returning. The default value is `20s`. The maximum value is `20s`.
 
 
+### `sqs.shutdown_grace_time` [_shutdown_grace_time]
+
+The duration that an SQS message processor will wait for a messages to arrive in the queue and be processed before allowing the input to terminate when a cancelation has been received. The default value is `20s`. It must not be negative.
```

Comment:
Does it also wait for new messages to arrive?
We would ideally want just the current batch or messages to finish processing/publication, right?

Reply:
That's right. Ideally, we'd shut the door, and keep an eye on who was inside and then shutdown after they had left, with a cut-off. In practice, we almost do that; there are two contexts in play, the context that is passed from to the input, `ctx`, and a child context that cancels `shutdown_grace_time` _after_ the parent has cancelled. The parent's cancellation stops new requests and the child stops the input `shutdown_grace_time` later.

Reply:
> The parent's cancellation stops new requests and the child stops the input shutdown_grace_time later.

Thanks for the clarification 👍🏼